tests/qml/helper.cpp

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#include "../../src/mozillavpn.h"
+#include "../../src/settingsholder.h"
+#include "l18nstrings.h"
+#include "helper.h"
+
+TestHelper::TestHelper() {
+  FeatureList::instance()->initialize();
+  m_whatsNewModel = new WhatsNewModel();
+}
+
+TestHelper* TestHelper::instance() {
+  static TestHelper* s_instance = new TestHelper();
+  return s_instance;
+}
+
+bool TestHelper::mainWindowLoadedCalled() {
+  // I don't understand why I need to call instance()
+  // here, would appreciate a c++ lesson and/or correction
+  // of my code.
+  return instance()->m_mainWindowLoadedCalled;
+}
+
+void TestHelper::setMainWindowLoadedCalled(bool val) {
+  m_mainWindowLoadedCalled = val;
+}
+
+void TestHelper::triggerInitializeGlean() {
+  emit MozillaVPN::instance()->initializeGlean();
+}
+
+void TestHelper::triggerRecordGleanEvent(const QString& event) {
+  emit MozillaVPN::instance()->recordGleanEvent(event);
+}
+
+void TestHelper::triggerSetGleanSourceTags(const QStringList& tags) {
+  emit MozillaVPN::instance()->setGleanSourceTags(tags);
+}
+
+void TestHelper::triggerAboutToQuit() {
+  emit MozillaVPN::instance()->aboutToQuit();
+}
+
+// The next two functions duplicate mozillavpn.h logic, but I'm not
+// sure how else to generate an "expected" value for tst_mainWindowGlean.qml
+// as it needs to be dynamic based on the platform people are running tests on.
+// At the least this should fail give someone pause for thought if they change
+// the mozillavpn.h logic.
+QString TestHelper::osVersion() { return QSysInfo::productVersion(); }
```

Comment:
this can be `const` too.

Reply:
Can you use MozillaVPN as a singleton and write this as:
`QString TestHelper::osVersion() const { return MozillaVPN::instance()->osVersion(); }` ?

Reply:
I don't think so because the MozillaVPN::instance() is the mocmozillavpn.cpp implementation, no?

If you know of a way of me being able to use the real mozillavpn instance that would be awesome.